src/ensembl/src/content/app/browser/browserHelper.ts

```diff
@@ -14,3 +20,47 @@ export function getChrLocationStr(
 
   return `${chrCode}:${startBp}-${endBp}`;
 }
+
+export const getRegionFieldErrorMessages = (
+  validationInfo: RegionValidationResponse | null
+) => {
+  // There is no validation to check if the response complains about region_code and genome_id not being valid.
+  // As the chance of this happening through the frontend is almost zero unless there are some spelling errors :/
+  try {
+    if (validationInfo) {
+      if (validationInfo.is_parseable === false) {
+        return RegionErrors.PARSE_ERROR;
+      } else if (validationInfo.region && !validationInfo.region.is_valid) {
+        return validationInfo.region.error_message;
+      } else if (validationInfo.start && !validationInfo.start.is_valid) {
+        return validationInfo.start.error_message;
+      } else if (validationInfo.end && !validationInfo.end.is_valid) {
+        return validationInfo.end.error_message;
+      }
+    }
+
+    return null;
+  } catch {
+    return RegionErrors.REQUEST_ERROR;
+  }
+};
```

Comment:
I am confused. Why the try-catch? What can possibly go wrong inside the try block that will produce an error that has to be caught?

Reply:
This was there when we didn't have a standard format for the  response and that caused some unexpected errors. Hence, the try-catch block. We probably don't need this anymore.